crypto/ctype.c

```diff
@@ -278,3 +286,38 @@ int ossl_ascii_isdigit(const char inchar) {
         return 1;
     return 0;
 }
+
+static locale_t loc;
+
+#ifndef FIPS_MODULE
+void *OPENSSL_strcasecmp_locale() {
+    return (void *)loc;
+}
+#endif
+
+int ossl_init_casecmp_int() {
+    loc = newlocale(LC_COLLATE_MASK, "C", (locale_t) 0);
+    return (loc == (locale_t) 0) ? 0 : 1;
+}
+
+void ossl_deinit_casecmp() {
+    freelocale(loc);
+}
+
+int ossl_strcasecmp(const char *s1, const char *s2)
+{
+# if defined(OPENSSL_SYS_WINDOWS)
+    return _stricmp(s1, s2);
+# else
+    return strcasecmp_l(s1, s2, (locale_t)OPENSSL_strcasecmp_locale());
+# endif
+}
+
+int ossl_strncasecmp(const char *s1, const char *s2, size_t n)
+{
+# if defined(OPENSSL_SYS_WINDOWS)
+    return _strnicmp(s1, s2, n);
+# else
+    return strncasecmp_l(s1, s2, n, (locale_t)OPENSSL_strcasecmp_locale());
+# endif
+}
```

Comment:
```suggestion
int ossl_strcasecmp(const char *s1, const char *s2)
{
# if defined(OPENSSL_SYS_WINDOWS)
    return _stricmp(s1, s2);
# elif defined(FIPS_MODULE)
    return strcasecmp(s1, s2); 
# else
    return strcasecmp_l(s1, s2, (locale_t)OPENSSL_strcasecmp_locale());
# endif
}

int ossl_strncasecmp(const char *s1, const char *s2, size_t n)
{
# if defined(OPENSSL_SYS_WINDOWS)
    return _strnicmp(s1, s2, n);
# elif defined(FIPS_MODULE)
    return strcnasecmp(s1, s2); 
# else
    return strncasecmp_l(s1, s2, n, (locale_t)OPENSSL_strcasecmp_locale());
# endif
}
```

Reply:
I'm not sure. Current version works in FIPS module, and if we need to compare something with I/i inside FIPS module in Turkish locale, we will get problems.